hilti/toolchain/src/compiler/global-optimizer.cc

```diff
@@ -212,12 +325,72 @@ struct Visitor : hilti::visitor::PreOrder<bool, Visitor> {
 
                 break;
             }
+
+            case Stage::PRUNE_DECLS:
+                // Nothing.
+                break;
         }
 
         return false;
     }
 
-    GlobalOptimizer::Hooks* _hooks = nullptr;
+    result_t operator()(const operator_::function::Call& call, position_t p) {
+        if ( ! call.hasOp0() )
+            return false;
+
+        auto id = call.op0().tryAs<expression::ResolvedID>();
+
+        if ( ! id )
+            return false;
+
+        auto module_id = id->id().sub(-2);
+        auto fn_id = id->id().sub(-1);
+
+        if ( module_id.empty() ) {
+            // Functions declared in this module do not include a module name in their ID.
+            if ( auto module = p.findParent<Module>() )
+                module_id = module->get().id();
+        }
```

Comment:
This works, though I've been doing elsewhere is a have a visitor method for `Module` and just save the `module_id` for later use.

Reply:
Let's leave this as is so the visitor methods are mostly concerned with extracting information or performing edits. If fear that performing additional work which will introduce ordering dependencies will make both the extraction and the uses harder to understand.